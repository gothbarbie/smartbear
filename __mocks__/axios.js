export const get = jest.fn()
export const put = jest.fn()
export default { get, put }
