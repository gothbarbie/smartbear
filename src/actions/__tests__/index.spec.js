import {
  getData,
  save,
} from '../index.js'

import mockStore from 'redux-mock-store'
import axios from 'axios'

describe('actions/index', () => {
  let store

  beforeEach(() => {
    store = mockStore()
  })

  describe('#getData', () => {
    axios.get.mockImplementationOnce(() =>
      Promise.resolve({
        data: [
          {
            virtualizationID: '1',
            apiType: 'REST',
            name: 'Pet Store',
            protocol: 'HTTP',
            port: '8081',
            running: false,
          },
          {
            virtualizationID: '2',
            apiType: 'WSDL',
            name: 'Currency Converter',
            protocol: 'HTTP',
            port: '8089',
            running: true,
          },
          {
            virtualizationID: '3',
            apiType: 'REST',
            name: 'PayStore',
            protocol: 'HTTPS',
            port: '8088',
            running: false,
          },
        ],
      })
    )

    store = mockStore({
      data: [
        {
          virtualizationID: '1',
          apiType: 'REST',
          name: 'Pet Store',
          protocol: 'HTTP',
          port: '8081',
          running: false,
        },
        {
          virtualizationID: '2',
          apiType: 'WSDL',
          name: 'Currency Converter',
          protocol: 'HTTP',
          port: '8089',
          running: true,
        },
        {
          virtualizationID: '3',
          apiType: 'REST',
          name: 'PayStore',
          protocol: 'HTTPS',
          port: '8088',
          running: false,
        },
      ],
    })

    it('fetches virtualizations', async () => {
      try {
        await store.dispatch(getData())

        expect(axios.get).toHaveBeenCalledWith(
          'http://localhost:8090/sv/v1/virtualizations'
        )
      } catch (error) {}
    })
  })

  describe('#save', () => {
    axios.put.mockImplementationOnce(() => Promise.resolve())

    it('saves new data', async () => {
      try {
        await store.dispatch(
          save({ id: 1, name: 'Pet Store', port: '8082', protocol: 'HTTP' })
        )

        expect(store.getActions()).toMatchSnapshot()
      } catch (error) {}
    })
  })
})
