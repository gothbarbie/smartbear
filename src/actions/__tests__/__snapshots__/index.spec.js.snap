// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`actions/index #save saves new data 1`] = `
Array [
  Object {
    "payload": Object {
      "id": 1,
      "name": "Pet Store",
      "port": "8082",
      "protocol": "HTTP",
    },
    "type": "data/UPDATE",
  },
]
`;
